session_0/item.rb

```diff
@@ -41,5 +54,18 @@ def discounted?
   # Note: If there are no items for category, stock price for category
   # should be zero.
   def self.stock_price_by_category(items)
+    stock_hash = {}
+    for id, val in CATEGORIES
```

Comment:
We prefer `each` over `for` loops in ruby - https://mixandgo.com/learn/how-to-use-ruby-each

Reply:
Ohk..I'll keep that in mind, Why is it so though like any time complexity measures?

For this code's alternative :-
CATEGORIES.each do |id, val|
    stock_hash[id] = 0
end

Reply:
Both each and for loops have the same complexity.

The problem with `for` loops is that the variables are accessible out of the scope and can lead to hard to fix bugs.

```ruby
for item in items
end

item # Refers to the last item in the array
```
```ruby
items.each do |item|
 ...
end

item # NilError
```

Reply:
That was a great info..Thanks :+1: